units/stats_task_end.go

```diff
@@ -52,51 +55,30 @@ func newTaskEndJob() *collectTaskEndDataJob {
 // NewCollectTaskEndDataJob logs information a task after it
 // completes. It also logs information about the total runtime and instance
 // type, which can be used to measure the cost of running a task.
-func NewCollectTaskEndDataJob(t *task.Task, h *host.Host) amboy.Job {
+func NewCollectTaskEndDataJob(t *task.Task, h *host.Host, p *pod.Pod) amboy.Job {
 	j := newTaskEndJob()
 	j.TaskID = t.Id
 	j.Execution = t.Execution
-	j.HostID = h.Id
+	var id string
+	if h != nil {
+		id = h.Id
+	} else {
+		id = p.ID
+	}
+	j.pod = p
 	j.task = t
 	j.host = h
-	j.SetID(fmt.Sprintf("%s.%s.%s.%d", collectTaskEndDataJobName, j.TaskID, j.HostID, job.GetNumber()))
+	j.SetID(fmt.Sprintf("%s.%s.%s.%d", collectTaskEndDataJobName, j.TaskID, id, job.GetNumber()))
 	j.SetPriority(-2)
 	return j
 }
 
 func (j *collectTaskEndDataJob) Run(ctx context.Context) {
 	defer j.MarkComplete()
 
-	var err error
-	if j.task == nil {
-		j.task, err = task.FindOneIdAndExecution(j.TaskID, j.Execution)
-		j.AddError(err)
-		if err != nil {
-			return
-		}
-	}
-	// The task was restarted before the job ran.
-	if j.task == nil {
-		j.task, err = task.FindOneOldByIdAndExecution(j.TaskID, j.Execution)
-		j.AddError(err)
-		if err != nil {
-			return
-		}
-	}
-	if j.task == nil {
-		j.AddError(errors.Errorf("Could not find task '%s'", j.TaskID))
-		return
-	}
-
-	if j.host == nil {
-		j.host, err = host.FindOneId(j.HostID)
-		j.AddError(err)
-		if err != nil {
-			return
-		}
-	}
-	if j.host == nil {
-		j.AddError(errors.Errorf("Could not find host '%s'", j.HostID))
+	isHostMode := j.host != nil
```

Comment:
This will always be nil because Amboy jobs in the remote queue are round-tripped through the DB, which zeroes out unexported fields such as `j.host`.